client/src/backoffice/AddressManagementPage.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { FC, useState, useEffect } from 'react';
+import classNames from 'classnames';
+
+/* Libraries */
+import ReactModal from 'react-modal';
+import { Formik, FormikActions, Form, Field, FieldProps, ErrorMessage } from 'formik';
+
+/* Helpers */
+import { GasPriceSchema } from '../lib/schemas';
+import { getSigners, setSigner, AdminAddress } from '../api';
+import { convertToGWEI, convertFromGWEI, convertToETH, reduceAddress } from '../lib/helpers';
+
+/* Components */
+import { SubmitButton } from '../components/SubmitButton';
+import { Loading } from '../components/Loading';
+
+/* Assets */
+import edit from '../images/edit.svg';
+
+type GasPriceFormValues = {
+  gasPrice: string;
+};
+
+ReactModal.setAppElement('#root');
+
+const AddressManagementPage: FC = () => {
+  const [isFetchingAddresses, setIsFetchingAddresses] = useState<null | boolean>(null);
+  const [addresses, setAddresses] = useState<null | AdminAddress[]>(null);
+  const [modalOpen, setModalOpen] = useState<boolean>(false);
+  const [selectedAddress, setSelectedAddress] = useState<null | AdminAddress>(null);
+
+  useEffect(() => {
+    fetchSigners();
+  }, []);
+
+  const fetchSigners = () => {
+    setIsFetchingAddresses(true);
+    setAddresses(null);
+
+    getSigners()
+      .then(addresses => {
+        if (!addresses) return;
+        setAddresses(addresses);
+      })
+      .catch(error => console.error(error))
+      .finally(() => setIsFetchingAddresses(false));
+  }
+
+  const handleFormSubmit = async (
+    values: GasPriceFormValues,
+    actions: FormikActions<GasPriceFormValues>
+  ) => {
+    if (!selectedAddress) return;
+    try {
+      actions.setStatus(null);
+      actions.setSubmitting(true);
+
+      const gasPriceInWEI = convertFromGWEI(values.gasPrice);
+      await setSigner(selectedAddress.id, gasPriceInWEI);
+      fetchSigners();
+      closeEditModal();
+    } catch (error) {
+      actions.setStatus({ ok: false, msg: `Gas price couldn't be changed` });
+    } finally {
+      actions.setSubmitting(false);
+    }
+  };
+
+  const openEditModal = (address: AdminAddress) => {
+    setModalOpen(true);
+    setSelectedAddress(address);
+  };
+
+  const closeEditModal = () => {
+    setModalOpen(false);
+    setSelectedAddress(null);
+  };
+
+
+  return (
+    <div className={'admin-table'}>
+      <h2>Admin addresses management</h2>
+      <div className={'row table-header'}>
+        <div className={'col-xs-1 center'}>#</div>
+        <div className={'col-xs-5'}>Address</div>
+        <div className={'col-xs-2'}>Role</div>
+        <div className={'col-xs-2 center'}>Balance (ETH)</div>
+        <div className={'col-xs-2 center'}>Gas Price (GWei)</div>
+      </div>
+      <div className={'admin-table-row'}>
+        {isFetchingAddresses && <Loading />}
+        {addresses && addresses.map((address, i) => {
+          return (
+            <div className={`row ${i % 2 === 0 ? 'even' : 'odd'}`} key={address.id}>
```

Comment:
Optionally you can target odd and even rows with a pseudo-selector:

Old:
```css
 .admin-table-row{
    .capitalize {
      text-transform: capitalize;
    }
    .even {
      background: white;
      border: 1px solid #d2d2d2;
    }
```

New:
```css
.admin-table-row {
  .row:nth-child(even) {
    background: white;
    border: 1px solid #d2d2d2;
   }
}
```

Reply:
Even better